the app crashes after `unlock the app` screen

Tested on Mi A1
```
I/Choreographer: Skipped 37 frames!  The application may be doing too much work on its main thread.
D/ProfileInstaller: Installing profile for io.parity.signer
D/navigation: initial unlock screen opened
D/navigation: main rust-handled screen navigation subgraph opened
D/strongbox available:: false
E/ENCRY: io.parity.signer.PolkadotVaultApp@3b8543b AndroidKeyStore MasterKey{keyAlias=_androidx_security_master_key_, isKeyStoreBacked=true}
W/o.parity.signe: Accessing hidden field Ljava/nio/Buffer;->address:J (light greylist, reflection)
W/AndroidKeystoreAesGcm: encountered a potentially transient KeyStore error, will wait and retry
    java.security.InvalidKeyException: Keystore operation failed
        at android.security.KeyStore.getInvalidKeyException(KeyStore.java:912)
        at android.security.KeyStore.getInvalidKeyException(KeyStore.java:937)
        at android.security.keystore.KeyStoreCryptoOperationUtils.getInvalidKeyExceptionForInit(KeyStoreCryptoOperationUtils.java:54)
        at android.security.keystore.KeyStoreCryptoOperationUtils.getExceptionForCipherInit(KeyStoreCryptoOperationUtils.java:89)
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.ensureKeystoreOperationInitialized(AndroidKeyStoreCipherSpiBase.java:265)
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineInit(AndroidKeyStoreCipherSpiBase.java:109)
        at javax.crypto.Cipher.tryTransformWithProvider(Cipher.java:2984)
        at javax.crypto.Cipher.tryCombinations(Cipher.java:2891)
        at javax.crypto.Cipher$SpiAndProviderUpdater.updateAndGetSpiAndProvider(Cipher.java:2796)
        at javax.crypto.Cipher.chooseProvider(Cipher.java:773)
        at javax.crypto.Cipher.init(Cipher.java:1143)
        at javax.crypto.Cipher.init(Cipher.java:1084)
        at com.google.crypto.tink.integration.android.AndroidKeystoreAesGcm.encryptInternal(AndroidKeystoreAesGcm.java:84)
        at com.google.crypto.tink.integration.android.AndroidKeystoreAesGcm.encrypt(AndroidKeystoreAesGcm.java:68)
        at com.google.crypto.tink.integration.android.AndroidKeystoreKmsClient.validateAead(AndroidKeystoreKmsClient.java:259)
        at com.google.crypto.tink.integration.android.AndroidKeystoreKmsClient.getAead(AndroidKeystoreKmsClient.java:175)
        at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.readOrGenerateNewMasterKey(AndroidKeysetManager.java:268)
        at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.build(AndroidKeysetManager.java:237)
        at androidx.security.crypto.EncryptedSharedPreferences.create(EncryptedSharedPreferences.java:165)
        at androidx.security.crypto.EncryptedSharedPreferences.create(EncryptedSharedPreferences.java:130)
        at androidx.security.crypto.EncryptedSharedPreferencesKt.EncryptedSharedPreferences(EncryptedSharedPreferences.kt:39)
        at io.parity.signer.domain.storage.SeedStorage.init(SeedStorage.kt:78)
        at io.parity.signer.domain.SharedViewModel.totalRefresh(SharedViewModel.kt:78)
        at io.parity.signer.domain.SharedViewModel$1.invoke(SharedViewModel.kt:26)
        at io.parity.signer.domain.SharedViewModel$1.invoke(SharedViewModel.kt:25)
        at io.parity.signer.domain.Authentication$authenticate$1.onAuthenticationSucceeded(Authentication.kt:99)
        at androidx.biometric.BiometricFragment$9.run(BiometricFragment.java:907)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6762)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: android.security.KeyStoreException: Invalid user authentication validity duration
        at android.security.KeyStore.getKeyStoreException(KeyStore.java:846)
        at android.security.KeyStore.getInvalidKeyException(KeyStore.java:937) 
        at android.security.keystore.KeyStoreCryptoOperationUtils.getInvalidKeyExceptionForInit(KeyStoreCryptoOperationUtils.java:54) 
        at android.security.keystore.KeyStoreCryptoOperationUtils.getExceptionForCipherInit(KeyStoreCryptoOperationUtils.java:89) 
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.ensureKeystoreOperationInitialized(AndroidKeyStoreCipherSpiBase.java:265) 
        at android.security.keystore.AndroidKeyStoreCipherSpiBase.engineInit(AndroidKeyStoreCipherSpiBase.java:109) 
        at javax.crypto.Cipher.tryTransformWithProvider(Cipher.java:2984) 
        at javax.crypto.Cipher.tryCombinations(Cipher.java:2891) 
        at javax.crypto.Cipher$SpiAndProviderUpdater.updateAndGetSpiAndProvider(Cipher.java:2796) 
        at javax.crypto.Cipher.chooseProvider(Cipher.java:773) 
        at javax.crypto.Cipher.init(Cipher.java:1143) 
        at javax.crypto.Cipher.init(Cipher.java:1084) 
        at com.google.crypto.tink.integration.android.AndroidKeystoreAesGcm.encryptInternal(AndroidKeystoreAesGcm.java:84) 
        at com.google.crypto.tink.integration.android.AndroidKeystoreAesGcm.encrypt(AndroidKeystoreAesGcm.java:68) 
        at com.google.crypto.tink.integration.android.AndroidKeystoreKmsClient.validateAead(AndroidKeystoreKmsClient.java:259) 
        at com.google.crypto.tink.integration.android.AndroidKeystoreKmsClient.getAead(AndroidKeystoreKmsClient.java:175) 
        at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.readOrGenerateNewMasterKey(AndroidKeysetManager.java:268) 
        at com.google.crypto.tink.integration.android.AndroidKeysetManager$Builder.build(AndroidKeysetManager.java:237) 
        at androidx.security.crypto.EncryptedSharedPreferences.create(EncryptedSharedPreferences.java:165) 
        at androidx.security.crypto.EncryptedSharedPreferences.create(EncryptedSharedPreferences.java:130) 
        at androidx.security.crypto.EncryptedSharedPreferencesKt.EncryptedSharedPreferences(EncryptedSharedPreferences.kt:39) 
        at io.parity.signer.domain.storage.SeedStorage.init(SeedStorage.kt:78) 
        at io.parity.signer.domain.SharedViewModel.totalRefresh(SharedViewModel.kt:78) 
        at io.parity.signer.domain.SharedViewModel$1.invoke(SharedViewModel.kt:26) 
        at io.parity.signer.domain.SharedViewModel$1.invoke(SharedViewModel.kt:25) 
        at io.parity.signer.domain.Authentication$authenticate$1.onAuthenticationSucceeded(Authentication.kt:99) 
        at androidx.biometric.BiometricFragment$9.run(BiometricFragment.java:907) 
        at android.os.Handler.handleCallback(Handler.java:873) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6762) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
D/AndroidRuntime: Shutting down VM
```

--- a/android/src/main/java/io/parity/signer/components/base/Buttons.kt
+++ b/android/src/main/java/io/parity/signer/components/base/Buttons.kt
@@ -26,11 +26,15 @@ fun PrimaryButtonWide(
 	label: String,
 	modifier: Modifier = Modifier,
 	isEnabled: Boolean = true,
+	activeBackground: Color = MaterialTheme.colors.pink500,
 	onClicked: Callback,
 ) {
 	PrimaryButton(
-		label, modifier.fillMaxWidth(1f),
-		isEnabled, onClicked
+		label = label,
+		modifier = modifier.fillMaxWidth(1f),
+		isEnabled = isEnabled,
+		activeBackground = activeBackground,
+		onClicked = onClicked
 	)
 }
 
@@ -39,6 +43,7 @@ fun PrimaryButton(
 	label: String,
 	modifier: Modifier = Modifier,
 	isEnabled: Boolean = true,
+	activeBackground: Color = MaterialTheme.colors.pink500,
 	onClicked: Callback
 ) {
 	Box(
@@ -51,7 +56,7 @@ fun PrimaryButton(
 				}
 			}
 			.background(
-				if (isEnabled) MaterialTheme.colors.pink500 else MaterialTheme.colors.primaryButtonDisabledBackground,
+				if (isEnabled) activeBackground else MaterialTheme.colors.primaryButtonDisabledBackground,
 				RoundedCornerShape(dimensionResource(id = R.dimen.buttonCornerRadius))
 			),
 		contentAlignment = Alignment.Center,
@@ -117,10 +122,15 @@ fun SecondaryButtonWide(
 	label: String,
 	modifier: Modifier = Modifier,
 	withBackground: Boolean = false,
+	textColor: Color = MaterialTheme.colors.textSecondary,
 	onClicked: Callback,
 ) {
 	SecondaryButton(
-		label, modifier.fillMaxWidth(), withBackground, onClicked,
+		label = label,
+		modifier = modifier.fillMaxWidth(),
+		withBackground = withBackground,
+		textColor = textColor,
+		onClicked = onClicked,
 	)
 }
 
@@ -129,6 +139,7 @@ private fun SecondaryButton(
 	label: String,
 	modifier: Modifier = Modifier,
 	withBackground: Boolean = false,
+	textColor: Color = MaterialTheme.colors.textSecondary,
 	onClicked: Callback,
 ) {
 	Column(
@@ -149,7 +160,7 @@ private fun SecondaryButton(
 	) {
 		Text(
 			text = label,
-			color = MaterialTheme.colors.textSecondary,
+			color = textColor,
 			style = SignerTypeface.TitleS,
 			maxLines = 1,
 		)

--- a/android/src/main/java/io/parity/signer/domain/SharedViewModel.kt
+++ b/android/src/main/java/io/parity/signer/domain/SharedViewModel.kt
@@ -90,11 +90,6 @@ class SharedViewModel() : ViewModel() {
 		ServiceLocator.uniffiInteractor.wasRustInitialized.value = true
 		networkExposedStateKeeper.updateAlertStateFromHistory()
 		navigator.navigate(Action.START)
-		if (allNames.isEmpty()) {
-			//workaround to hide create new bottom sheet while #1618 is not merged
-			//https://github.com/paritytech/parity-signer/pull/1618
-			navigator.navigate(Action.GO_BACK)
-		}
 	}
 
 	/**

--- a/android/src/main/java/io/parity/signer/domain/storage/SeedStorage.kt
+++ b/android/src/main/java/io/parity/signer/domain/storage/SeedStorage.kt
@@ -57,12 +57,12 @@ class SeedStorage {
 			MasterKey.Builder(appContext)
 				.setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
 				.setRequestStrongBoxBacked(true)
-				.setUserAuthenticationRequired(true)
+				.setUserAuthenticationRequired(true, MasterKey.getDefaultAuthenticationValidityDurationSeconds())
 				.build()
 		} else {
 			MasterKey.Builder(appContext)
 				.setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
-				.setUserAuthenticationRequired(true)
+				.setUserAuthenticationRequired(true, MasterKey.getDefaultAuthenticationValidityDurationSeconds())
 				.build()
 		}
 

--- a/android/src/main/java/io/parity/signer/screens/keysets/KeySetsNavSubgraph.kt
+++ b/android/src/main/java/io/parity/signer/screens/keysets/KeySetsNavSubgraph.kt
@@ -3,6 +3,8 @@ package io.parity.signer.screens.keysets
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.statusBarsPadding
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.State
 import androidx.compose.ui.Modifier
 import androidx.navigation.compose.NavHost
@@ -13,6 +15,7 @@ import io.parity.signer.domain.KeySetsSelectModel
 import io.parity.signer.domain.Navigator
 import io.parity.signer.screens.keysets.export.KeySetsExportScreenFull
 import io.parity.signer.ui.BottomSheetWrapperRoot
+import io.parity.signer.uniffi.Action
 
 /**
  * Navigation Subgraph with compose nav controller for those Key Set screens which are not part of general
@@ -38,6 +41,13 @@ fun KeySetsNavSubgraph(
 					localNavigator = navController,
 					networkState = networkState,
 				)
+			}
+			LaunchedEffect(key1 = Unit,) {
+				if (model.keys.isEmpty()) {
+					//workaround to hide create new bottom sheet while #1618 is not merged
+					//https://github.com/paritytech/parity-signer/pull/1618
+					rootNavigator.navigate(Action.GO_BACK)
+				}
 			}
 		}
 		composable(KeySetsNavSubgraph.homeMenu) {

--- a/android/src/main/java/io/parity/signer/screens/keysets/KeySetsNavSubgraph.kt
+++ b/android/src/main/java/io/parity/signer/screens/keysets/KeySetsNavSubgraph.kt
@@ -50,9 +50,7 @@ fun KeySetsNavSubgraph(
 				)
 			}
 			BottomSheetWrapperRoot(onClosedAction = {
-				navController.navigate(
-					KeySetsNavSubgraph.home
-				)
+				navController.popBackStack(KeySetsNavSubgraph.home, false)
 			}) {
 				KeySetsMenuBottomSheet(navigator = navController)
 			}
@@ -60,7 +58,9 @@ fun KeySetsNavSubgraph(
 		composable(KeySetsNavSubgraph.export) {
 			KeySetsExportScreenFull(
 				model = model,
-				onClose = { navController.navigate(KeySetsNavSubgraph.home) },
+				onClose = {
+					navController.popBackStack(KeySetsNavSubgraph.home, false)
+				},
 			)
 		}
 	}

--- a/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/EachStartSubgraphScreen.kt
+++ b/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/EachStartSubgraphScreen.kt
@@ -18,6 +18,9 @@ import androidx.lifecycle.viewmodel.compose.viewModel
 import androidx.navigation.NavGraphBuilder
 import androidx.navigation.NavHostController
 import androidx.navigation.compose.composable
+import io.parity.signer.domain.Callback
+import io.parity.signer.domain.NetworkState
+import io.parity.signer.domain.isDbCreatedAndOnboardingPassed
 import io.parity.signer.screens.onboarding.eachstartchecks.airgap.AirgapScreen
 import io.parity.signer.screens.onboarding.eachstartchecks.rootcheck.RootExposedScreen
 import io.parity.signer.screens.onboarding.eachstartchecks.screenlock.SetScreenLockScreen
@@ -29,14 +32,22 @@ fun NavGraphBuilder.enableEachStartAppFlow(globalNavController: NavHostControlle
 		val viewModel: EachStartViewModel = viewModel()
 		val context: Context = LocalContext.current
 
+		val goToNextFlow: Callback = {
+			globalNavController.navigate(MainGraphRoutes.initialUnlockRoute) {
+				popUpTo(0)
+			}
+		}
+
 		val state = remember {
 			mutableStateOf(
 				if (viewModel.isDeviceRooted()) {
 					EachStartSubgraphScreenSteps.ROOT_EXPOSED
-				} else if (viewModel.checkIsAuthPossible(context)) {
+				} else if (!viewModel.checkIsAuthPossible(context)) {
+					EachStartSubgraphScreenSteps.SET_SCREEN_LOCK_BLOCKER
+				} else if (viewModel.networkState.value == NetworkState.Active || !context.isDbCreatedAndOnboardingPassed()){
 					EachStartSubgraphScreenSteps.AIR_GAP
 				} else {
-					EachStartSubgraphScreenSteps.SET_SCREEN_LOCK_BLOCKER
+					goToNextFlow()
 				}
 			)
 		}
@@ -71,9 +82,7 @@ fun NavGraphBuilder.enableEachStartAppFlow(globalNavController: NavHostControlle
 				EachStartSubgraphScreenSteps.AIR_GAP -> {
 					AirgapScreen {
 						//go to next screen
-						globalNavController.navigate(MainGraphRoutes.initialUnlockRoute) {
-							popUpTo(0)
-						}
+						goToNextFlow()
 					}
 				}
 			}

--- a/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/EachStartViewModel.kt
+++ b/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/EachStartViewModel.kt
@@ -15,10 +15,15 @@ import kotlinx.coroutines.flow.map
 
 class EachStartViewModel : ViewModel() {
 
+	private val networkExposedStateKeeper =
+		ServiceLocator.networkExposedStateKeeper
+
 	fun checkIsAuthPossible(context: Context): Boolean = Authentication.canAuthenticate(context)
 
 	fun isDeviceRooted(): Boolean {
 		return RootUtils.isDeviceRooted()
 	}
+
+	val networkState: StateFlow<NetworkState> = networkExposedStateKeeper.airGapModeState
 
 }

--- a/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/airgap/AirgapScreen.kt
+++ b/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/airgap/AirgapScreen.kt
@@ -38,7 +38,6 @@ import io.parity.signer.ui.theme.*
 fun AirgapScreen(
 	onCta: Callback,
 ) {
-
 	val viewModel = viewModel<AirGapViewModel>()
 	val state = viewModel.state.collectAsState()
 	DisposableEffect(Unit) {

--- a/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/airgap/AirgapScreen.kt
+++ b/android/src/main/java/io/parity/signer/screens/onboarding/eachstartchecks/airgap/AirgapScreen.kt
@@ -1,9 +1,7 @@
 package io.parity.signer.screens.onboarding.eachstartchecks.airgap
 
 import android.content.res.Configuration
-import androidx.compose.foundation.BorderStroke
-import androidx.compose.foundation.Image
-import androidx.compose.foundation.background
+import androidx.compose.foundation.*
 import androidx.compose.foundation.layout.*
 import androidx.compose.foundation.shape.CircleShape
 import androidx.compose.foundation.shape.RoundedCornerShape
@@ -63,84 +61,88 @@ private fun AirgapScreen(
 	onCablesConfirmCheckbox: Callback,
 	onCta: Callback,
 ) {
-	Column(horizontalAlignment = Alignment.CenterHorizontally) {
-
-		Text(
+	Column() {
+		Column(
 			modifier = Modifier
-				.fillMaxWidth(1f)
-				.padding(horizontal = 24.dp, vertical = 12.dp),
-			text = stringResource(R.string.airgap_onboarding_title),
-			color = MaterialTheme.colors.primary,
-			style = SignerTypeface.TitleL,
-			textAlign = TextAlign.Center,
-		)
-		Text(
-			modifier = Modifier
-				.fillMaxWidth(1f)
-				.padding(horizontal = 24.dp),
-			text = stringResource(R.string.airgap_onboarding_subtitle),
-			color = MaterialTheme.colors.textTertiary,
-			style = SignerTypeface.BodyL,
-			textAlign = TextAlign.Center,
-		)
-
-		Surface(
-			shape = RoundedCornerShape(dimensionResource(id = R.dimen.innerFramesCornerRadius)),
-			border = BorderStroke(1.dp, color = MaterialTheme.colors.fill12),
-			color = MaterialTheme.colors.fill6,
-			modifier = Modifier.padding(16.dp)
+				.verticalScroll(rememberScrollState())
+				.weight(1f, true)
 		) {
-			Column(
-				horizontalAlignment = Alignment.CenterHorizontally,
-			) {
-				AirgapItem(AirgapItemType.AIRPLANE_MODE, state.airplaneModeEnabled)
-				SignerDivider(modifier = Modifier.padding(start = 40.dp))
-				AirgapItem(AirgapItemType.WIFI, state.wifiDisabled)
-				SignerDivider(modifier = Modifier.padding(start = 40.dp))
-				AirgapItem(AirgapItemType.BLUETOOTH, state.bluetoothDisabled)
-			}
-		}
+			Text(
+				modifier = Modifier
+					.fillMaxWidth(1f)
+					.padding(horizontal = 24.dp, vertical = 12.dp),
+				text = stringResource(R.string.airgap_onboarding_title),
+				color = MaterialTheme.colors.primary,
+				style = SignerTypeface.TitleL,
+				textAlign = TextAlign.Center,
+			)
+			Text(
+				modifier = Modifier
+					.fillMaxWidth(1f)
+					.padding(horizontal = 24.dp),
+				text = stringResource(R.string.airgap_onboarding_subtitle),
+				color = MaterialTheme.colors.textTertiary,
+				style = SignerTypeface.BodyL,
+				textAlign = TextAlign.Center,
+			)
 
-		Surface(
-			shape = RoundedCornerShape(dimensionResource(id = R.dimen.innerFramesCornerRadius)),
-			border = BorderStroke(1.dp, color = MaterialTheme.colors.fill12),
-			color = MaterialTheme.colors.fill6,
-			modifier = Modifier.padding(16.dp)
-		) {
-			Column(
-				horizontalAlignment = Alignment.CenterHorizontally,
+			Surface(
+				shape = RoundedCornerShape(dimensionResource(id = R.dimen.innerFramesCornerRadius)),
+				border = BorderStroke(1.dp, color = MaterialTheme.colors.fill12),
+				color = MaterialTheme.colors.fill6,
+				modifier = Modifier.padding(16.dp)
 			) {
-				Row(
-					verticalAlignment = Alignment.CenterVertically,
-					modifier = Modifier.padding(16.dp),
+				Column(
+					horizontalAlignment = Alignment.CenterHorizontally,
 				) {
-					Image(
-						imageVector = Icons.Filled.Cable,
-						contentDescription = null,
-						colorFilter = ColorFilter.tint(MaterialTheme.colors.textTertiary),
-						modifier = Modifier
-							.size(24.dp)
-					)
-					Text(
-						text = stringResource(R.string.airgap_onboarding_disconnect_cable_header),
-						color = MaterialTheme.colors.textTertiary,
-						style = SignerTypeface.TitleS,
-						modifier = Modifier
-							.padding(horizontal = 16.dp, vertical = 14.dp)
-							.weight(1f)
-					)
+					AirgapItem(AirgapItemType.AIRPLANE_MODE, state.airplaneModeEnabled)
+					SignerDivider(modifier = Modifier.padding(start = 40.dp))
+					AirgapItem(AirgapItemType.WIFI, state.wifiDisabled)
+					SignerDivider(modifier = Modifier.padding(start = 40.dp))
+					AirgapItem(AirgapItemType.BLUETOOTH, state.bluetoothDisabled)
 				}
-				SignerDivider()
-				CheckboxWithText(
-					checked = state.cablesDisconnected,
-					text = stringResource(R.string.airgap_onboarding_disconnect_cable_checkbox_description),
-					modifier = Modifier.padding(16.dp),
-				) { newIsChecked ->
-					onCablesConfirmCheckbox()
+			}
+
+			Surface(
+				shape = RoundedCornerShape(dimensionResource(id = R.dimen.innerFramesCornerRadius)),
+				border = BorderStroke(1.dp, color = MaterialTheme.colors.fill12),
+				color = MaterialTheme.colors.fill6,
+				modifier = Modifier.padding(16.dp)
+			) {
+				Column(
+					horizontalAlignment = Alignment.CenterHorizontally,
+				) {
+					Row(
+						verticalAlignment = Alignment.CenterVertically,
+						modifier = Modifier.padding(16.dp),
+					) {
+						Image(
+							imageVector = Icons.Filled.Cable,
+							contentDescription = null,
+							colorFilter = ColorFilter.tint(MaterialTheme.colors.textTertiary),
+							modifier = Modifier
+								.size(24.dp)
+						)
+						Text(
+							text = stringResource(R.string.airgap_onboarding_disconnect_cable_header),
+							color = MaterialTheme.colors.textTertiary,
+							style = SignerTypeface.TitleS,
+							modifier = Modifier
+								.padding(horizontal = 16.dp, vertical = 14.dp)
+								.weight(1f)
+						)
+					}
+					SignerDivider()
+					CheckboxWithText(
+						checked = state.cablesDisconnected,
+						text = stringResource(R.string.airgap_onboarding_disconnect_cable_checkbox_description),
+						modifier = Modifier.padding(16.dp),
+					) { newIsChecked ->
+						onCablesConfirmCheckbox()
+					}
 				}
 			}
 		}
-		Spacer(modifier = Modifier.weight(1f))
 		PrimaryButtonWide(
 			label = stringResource(R.string.button_next),
 			modifier = Modifier.padding(24.dp),
@@ -155,8 +157,7 @@ data class AirGapScreenState(
 	val wifiDisabled: Boolean,
 	val bluetoothDisabled: Boolean,
 	val cablesDisconnected: Boolean = false, //false default
-) {
-}
+)
 
 private fun AirGapScreenState.isReadyToProceed() =
 	airplaneModeEnabled && wifiDisabled
@@ -255,3 +256,24 @@ private fun PreviewAirgapScreen() {
 }
 
 
+@Preview(
+	name = "light", group = "themes", uiMode = Configuration.UI_MODE_NIGHT_NO,
+	showBackground = true, backgroundColor = 0xFFFFFFFF,
+)
+@Preview(
+	name = "dark", group = "themes", uiMode = Configuration.UI_MODE_NIGHT_YES,
+	showBackground = true, backgroundColor = 0xFF000000,
+)
+@Composable
+private fun PreviewAirgapScreenSmall() {
+	Box(modifier = Modifier.size(width = 250.dp, height = 450.dp)) {
+		SignerNewTheme() {
+			val state = AirGapScreenState(
+				airplaneModeEnabled = true,
+				wifiDisabled = false,
+				bluetoothDisabled = true
+			)
+			AirgapScreen(state, {}, {})
+		}
+	}
+}

--- a/android/src/main/java/io/parity/signer/screens/settings/SettingsScreen.kt
+++ b/android/src/main/java/io/parity/signer/screens/settings/SettingsScreen.kt
@@ -2,10 +2,7 @@ package io.parity.signer.screens.settings
 
 import android.content.res.Configuration
 import androidx.compose.foundation.*
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Row
-import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.*
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Text
 import androidx.compose.material.icons.Icons
@@ -15,10 +12,13 @@ import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.ColorFilter
 import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import io.parity.signer.R
 import io.parity.signer.alerts.AndroidCalledConfirm
+import io.parity.signer.components.base.BottomSheetConfirmDialog
+import io.parity.signer.components.base.RowButtonsBottomSheet
 import io.parity.signer.components.base.ScreenHeader
 import io.parity.signer.components.documents.PpScreen
 import io.parity.signer.components.documents.TosScreen

--- a/android/src/main/java/io/parity/signer/screens/settings/SettingsScreen.kt
+++ b/android/src/main/java/io/parity/signer/screens/settings/SettingsScreen.kt
@@ -2,23 +2,28 @@ package io.parity.signer.screens.settings
 
 import android.content.res.Configuration
 import androidx.compose.foundation.*
-import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
 import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Text
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.filled.ChevronRight
-import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.ColorFilter
 import androidx.compose.ui.res.stringResource
-import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
 import io.parity.signer.R
-import io.parity.signer.alerts.AndroidCalledConfirm
-import io.parity.signer.components.base.BottomSheetConfirmDialog
-import io.parity.signer.components.base.RowButtonsBottomSheet
 import io.parity.signer.components.base.ScreenHeader
 import io.parity.signer.components.documents.PpScreen
 import io.parity.signer.components.documents.TosScreen
@@ -29,6 +34,7 @@ import io.parity.signer.domain.Callback
 import io.parity.signer.domain.EmptyNavigator
 import io.parity.signer.domain.Navigator
 import io.parity.signer.domain.NetworkState
+import io.parity.signer.ui.BottomSheetWrapperRoot
 import io.parity.signer.ui.theme.*
 import io.parity.signer.uniffi.Action
 
@@ -38,50 +44,73 @@ import io.parity.signer.uniffi.Action
  * all subsequent interactions should be in modals or drop-down menus
  */
 @Composable
-fun SettingsScreen(
+fun SettingsScreenSubgraph(
 	rootNavigator: Navigator,
 	isStrongBoxProtected: Boolean,
 	appVersion: String,
 	wipeToFactory: Callback,
 	networkState: State<NetworkState?>
 ) {
-	var confirmWipe by remember { mutableStateOf(false) }
-	var settingsState by remember { mutableStateOf(SettingsState.GENERAL_SCREEN) }
+	val navController = rememberNavController()
+	NavHost(
+		navController = navController,
+		startDestination = SettingsScreenSubgraph.home,
+	) {
 
-	when (settingsState) {
-		SettingsState.GENERAL_SCREEN ->
+		composable(SettingsScreenSubgraph.home) {
 			SettingsScreenGeneralView(
 				rootNavigator,
-				onWipeData = { confirmWipe = true },
-				onShowTerms = { settingsState = SettingsState.TERMS_OF_SERVICE },
-				onShowPrivacyPolicy = { settingsState = SettingsState.PRIVACY_POLICY },
+				onWipeData = { navController.navigate(SettingsScreenSubgraph.wipeConformation) },
+				onShowTerms = { navController.navigate(SettingsScreenSubgraph.terms) },
+				onShowPrivacyPolicy = { navController.navigate(SettingsScreenSubgraph.privacyPolicy) },
 				isStrongBoxProtected,
 				appVersion,
 				networkState
 			)
-		SettingsState.TERMS_OF_SERVICE ->
-			TosScreen(onBack = { settingsState = SettingsState.GENERAL_SCREEN })
-		SettingsState.PRIVACY_POLICY ->
-			PpScreen(onBack = { settingsState = SettingsState.GENERAL_SCREEN })
-	}
+		}
+		composable(SettingsScreenSubgraph.wipeConformation) {
+			SettingsScreenGeneralView(
+				rootNavigator,
+				onWipeData = { navController.navigate(SettingsScreenSubgraph.wipeConformation) },
+				onShowTerms = { navController.navigate(SettingsScreenSubgraph.terms) },
+				onShowPrivacyPolicy = { navController.navigate(SettingsScreenSubgraph.privacyPolicy) },
+				isStrongBoxProtected,
+				appVersion,
+				networkState
+			)
 
-	AndroidCalledConfirm(
-		show = confirmWipe,
-		header = "Wipe ALL data?",
-		text = "Factory reset the Signer app. This operation can not be reverted!",
-		back = { confirmWipe = false },
-		forward = { wipeToFactory() },
-		backText = "Cancel",
-		forwardText = "Wipe"
-	)
-
-	DisposableEffect(key1 = Unit) {
-		onDispose { settingsState = SettingsState.GENERAL_SCREEN }
+			BottomSheetWrapperRoot(onClosedAction = {
+				navController.popBackStack(SettingsScreenSubgraph.home, false)
+			}) {
+				SettingsWipeAllConfirmation(
+					onCancel = {
+						navController.popBackStack(
+							SettingsScreenSubgraph.home,
+							false
+						)
+					},
+					onWipe = wipeToFactory
+				)
+			}
+		}
+		composable(SettingsScreenSubgraph.terms) {
+			TosScreen(onBack = {
+				navController.popBackStack(SettingsScreenSubgraph.home, false)
+			})
+		}
+		composable(SettingsScreenSubgraph.privacyPolicy) {
+			PpScreen(onBack = {
+				navController.popBackStack(SettingsScreenSubgraph.home, false)
+			})
+		}
 	}
 }
 
-private enum class SettingsState {
-	GENERAL_SCREEN, TERMS_OF_SERVICE, PRIVACY_POLICY,
+private object SettingsScreenSubgraph {
+	const val home = "settings_home"
+	const val wipeConformation = "setting_whipe_all_conformation"
+	const val terms = "settings_terms_of_service"
+	const val privacyPolicy = "settings_privacy_polcy"
 }
 
 @Composable
@@ -120,24 +149,22 @@ private fun SettingsScreenGeneralView(
 				)
 				Text(
 					text = stringResource(
-						R.string.settings_hardware_key,
-						isStrongBoxProtected.toString()
+						R.string.settings_hardware_key, isStrongBoxProtected.toString()
 					),
 					style = SignerTypeface.BodyM,
 					color = MaterialTheme.colors.textSecondary,
-					modifier = Modifier
-						.padding(horizontal = 24.dp, vertical = 16.dp)
+					modifier = Modifier.padding(horizontal = 24.dp, vertical = 16.dp)
 				)
 				Text(
 					text = stringResource(R.string.settings_version, appVersion),
 					style = SignerTypeface.BodyM,
 					color = MaterialTheme.colors.textSecondary,
-					modifier = Modifier
-						.padding(horizontal = 24.dp)
+					modifier = Modifier.padding(horizontal = 24.dp)
 				)
 			}
 			ExposedIcon(
-				networkState = networkState, navigator = rootNavigator,
+				networkState = networkState,
+				navigator = rootNavigator,
 				modifier = Modifier
 					.align(Alignment.BottomEnd)
 					.padding(end = 16.dp, bottom = 16.dp)
@@ -191,11 +218,13 @@ internal fun SettingsElement(
 private fun PreviewSettingsScreen() {
 	SignerNewTheme {
 		val state = remember { mutableStateOf(NetworkState.Past) }
-		SettingsScreen(
+		SettingsScreenGeneralView(
 			rootNavigator = EmptyNavigator(),
+			onWipeData = {},
+			onShowTerms = {},
+			onShowPrivacyPolicy = {},
 			isStrongBoxProtected = false,
 			appVersion = "0.6.1",
-			wipeToFactory = {},
 			networkState = state,
 		)
 	}

--- a/android/src/main/java/io/parity/signer/screens/settings/SettingsWipeConfirmation.kt
+++ b/android/src/main/java/io/parity/signer/screens/settings/SettingsWipeConfirmation.kt
@@ -1,0 +1,102 @@
+package io.parity.signer.screens.settings
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.*
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import io.parity.signer.R
+import io.parity.signer.components.base.PrimaryButtonWide
+import io.parity.signer.components.base.RowButtonsBottomSheet
+import io.parity.signer.components.base.SecondaryButtonWide
+import io.parity.signer.domain.Callback
+import io.parity.signer.ui.theme.SignerNewTheme
+import io.parity.signer.ui.theme.SignerTypeface
+import io.parity.signer.ui.theme.pink300
+import io.parity.signer.ui.theme.textSecondary
+
+
+@Composable
+fun SettingsWipeAllConfirmation(
+	onWipe: Callback,
+	onCancel: Callback,
+) {
+	val sidePadding = 24.dp
+	Column(
+		modifier = Modifier
+			.fillMaxWidth()
+			.padding(start = sidePadding, end = sidePadding, top = 32.dp),
+	) {
+
+		Text(
+			modifier = Modifier.fillMaxWidth(1f),
+			text = stringResource(R.string.settings_confirmation_wipe_all_title),
+			color = MaterialTheme.colors.primary,
+			style = SignerTypeface.TitleL,
+		)
+
+		val warning =
+			stringResource(R.string.settings_confirmation_wipe_all_message_highlited_warning).uppercase()
+		val message =
+			stringResource(R.string.settings_confirmation_wipe_all_message, warning)
+		val startPosition = message.indexOf(warning)
+		val spanStyles = listOf(
+			AnnotatedString.Range(
+				SpanStyle(
+					fontWeight = FontWeight.Bold,
+					color = MaterialTheme.colors.pink300
+				),
+				start = startPosition,
+				end = startPosition + warning.length
+			)
+		)
+		Text(
+			modifier = Modifier
+				.fillMaxWidth(1f)
+				.padding(
+					top = 16.dp, bottom = 24.dp,
+				),
+			text = AnnotatedString(message, spanStyles),
+			color = MaterialTheme.colors.textSecondary,
+			style = SignerTypeface.BodyL,
+		)
+		PrimaryButtonWide(
+			label = stringResource(R.string.settings_confirmation_wipe_all_message_yes_label),
+			activeBackground = Color.Red,
+			onClicked = onWipe
+		)
+		SecondaryButtonWide(
+			label = stringResource(R.string.generic_cancel),
+			textColor = MaterialTheme.colors.primary,
+			onClicked = onCancel,
+		)
+		Spacer(modifier = Modifier.padding(bottom = 24.dp))
+	}
+}
+
+
+@Preview(
+	name = "light", group = "general", uiMode = Configuration.UI_MODE_NIGHT_NO,
+	showBackground = true, backgroundColor = 0xFFFFFFFF,
+)
+@Preview(
+	name = "dark", group = "general",
+	uiMode = Configuration.UI_MODE_NIGHT_YES,
+	showBackground = true, backgroundColor = 0xFF000000,
+)
+@Composable
+private fun PreviewSettingsWipeAllConfirmation() {
+	SignerNewTheme {
+		SettingsWipeAllConfirmation({}, {})
+	}
+}

--- a/android/src/main/java/io/parity/signer/screens/settings/SettingsWipeConfirmation.kt
+++ b/android/src/main/java/io/parity/signer/screens/settings/SettingsWipeConfirmation.kt
@@ -11,13 +11,13 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.res.stringResource
-import androidx.compose.ui.text.*
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.SpanStyle
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import io.parity.signer.R
 import io.parity.signer.components.base.PrimaryButtonWide
-import io.parity.signer.components.base.RowButtonsBottomSheet
 import io.parity.signer.components.base.SecondaryButtonWide
 import io.parity.signer.domain.Callback
 import io.parity.signer.ui.theme.SignerNewTheme

--- a/android/src/main/java/io/parity/signer/ui/rustnavigationselectors/ScreenSelectorsNew.kt
+++ b/android/src/main/java/io/parity/signer/ui/rustnavigationselectors/ScreenSelectorsNew.kt
@@ -31,7 +31,7 @@ import io.parity.signer.screens.logs.LogsMenu
 import io.parity.signer.screens.logs.LogsScreen
 import io.parity.signer.screens.logs.toLogsScreenModel
 import io.parity.signer.screens.scan.ScanNavSubgraph
-import io.parity.signer.screens.settings.SettingsScreen
+import io.parity.signer.screens.settings.SettingsScreenSubgraph
 import io.parity.signer.ui.BottomSheetWrapperRoot
 import io.parity.signer.ui.theme.SignerNewTheme
 import io.parity.signer.uniffi.Action
@@ -89,7 +89,7 @@ fun CombinedScreensSelector(
 			}
 		is ScreenData.Settings ->
 			Box(modifier = Modifier.statusBarsPadding()) {
-				SettingsScreen(
+				SettingsScreenSubgraph(
 					rootNavigator = rootNavigator,
 					isStrongBoxProtected = sharedViewModel.seedStorage.isStrongBoxProtected,
 					appVersion = sharedViewModel.getAppVersion(),

--- a/android/src/main/res/values/strings.xml
+++ b/android/src/main/res/values/strings.xml
@@ -304,7 +304,10 @@
 	<string name="airgap_onboarding_bluetooth_header">Turn off Bluetooth</string>
 	<string name="airgap_onboarding_airplane_mode_header">Enable Airplane Mode</string>
 	<string name="airgap_onboarding_disconnect_cable_checkbox_description">I confirm that all Cables are Disconnected</string>
-
+	<string name="settings_confirmation_wipe_all_message">Remove all data and set general verifier information blank. \n\n%1$s This operation cannot be reversed so do not proceed unless you are sure you wish to erase important verification information. In most cases, there is no need to wipe all data. Using this feature inappropriately may lead to loss of funds!</string>
+	<string name="settings_confirmation_wipe_all_message_highlited_warning">Warning:</string>
+	<string name="settings_confirmation_wipe_all_title">Wipe All Data?</string>
+	<string name="settings_confirmation_wipe_all_message_yes_label">I Understand, Remove all Data</string>
 
 </resources>
 

--- a/build.gradle
+++ b/build.gradle
@@ -12,7 +12,7 @@ buildscript {
 		maven { url "https://plugins.gradle.org/m2/" }
 	}
 	dependencies {
-		classpath 'com.android.tools.build:gradle:7.4.1'
+		classpath 'com.android.tools.build:gradle:7.4.2'
 		classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
 		classpath "org.mozilla.rust-android-gradle:plugin:0.9.3"
 		// uniffi wants some build-time specific info from JNA

